### PR TITLE
feat(condo): DOMA-10801 ResetUserLimitAction

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
We need a tool that will allow QA to reset SMS limit counters by himself

### CODE REVIEW

**Review by toplenboren — APPROVED** [review]

**Comment by toplenboren** [comment]
Best to also check that IP address is not reserved one (so that I cannot use 127.0.0.1 as arg)

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/user/access/_internalResetSMSDayLimitCountersService.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Generated by `createservice user._internalResetSMSDayLimitCountersService`
+ */
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+
+async function can_internalResetSMSDayLimitCounters ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return true
```
Comment (SavelevMatthew) [inline apps/condo/domains/user/access/_internalResetSMSDayLimitCountersService.js:10]: There's a lot of people in support right now, maybe we should add directAccessRight here instead?

File: apps/condo/domains/user/utils/serverSchema/guards.js [hunk 2]
```diff
@@ -77,6 +77,21 @@ class RedisGuard {
...
+        const key = `${this.counterPrefix}${variable}`
+        return await this.redis.get(key)
+    }
+
+    async checkCounterExistence (variable) {
+        const key = `${this.counterPrefix}${variable}`
+        return await this.redis.exists(key)
+    }
+
+    async resetCounterToOne (variable) {
```
Comment (SavelevMatthew) [inline apps/condo/domains/user/utils/serverSchema/guards.js:90]: I suggest to delete it, why 1?

**Review by SavelevMatthew — COMMENTED** [review]
I don't really like using the default prefix, `guard_lock:` looks too generic. Maybe we should change it to something like `guard_lock:auth: `?

And resetting to 1 is strange, seems like you're copying someones local wrong-implemented script :)

File: apps/condo/domains/user/access/_internalResetSMSDayLimitCountersService.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/user/access/_internalResetSMSDayLimitCountersService.js:10]: Who will be the manager of the new UserRightSet model (connect it to the User models and disconnect sometimes)?

Comment (SavelevMatthew) [reply apps/condo/domains/user/access/_internalResetSMSDayLimitCountersService.js:10]: The head of support I guess

File: apps/condo/domains/user/utils/clientSchema/ResetUserLimitAction.ts
Comment (SavelevMatthew) [inline apps/condo/domains/user/utils/clientSchema/ResetUserLimitAction.ts:1]: Remove this file :)

Can you also remove it from template as well?

File: apps/condo/domains/user/utils/clientSchema/index.ts [hunk 3]
```diff
@@ -8,4 +8,5 @@ export * as ConfirmPhoneAction from './ConfirmPhoneAction'
 export * as OidcClient from './OidcClient'
 export * as ExternalTokenAccessRight from './ExternalTokenAccessRight'
 export * as UserRightsSet from './UserRightsSet'
+export * as ResetUserLimitAction from './ResetUserLimitAction'
```
Comment (SavelevMatthew) [inline apps/condo/domains/user/utils/clientSchema/index.ts:11]: remove

File: apps/condo/domains/user/utils/directAccess/config.js [hunk 4]
```diff
@@ -59,6 +59,7 @@ const DIRECT_ACCESS_AVAILABLE_SCHEMAS = {
         'OidcClient',
         { schemaName: 'User', readonly: true },
         'UserRightsSet',
+        'ResetUserLimitAction',
```
Comment (SavelevMatthew) [inline apps/condo/domains/user/utils/directAccess/config.js:62]: Keep alphabetical sort inside domain please

**Review by SavelevMatthew — APPROVED** [review]

**Pull request merged**